pbrd/pbr_vty.c

```diff
@@ -0,0 +1,632 @@
+      "v6 Address\n"
+      "Interface to use\n"
+      "If the nexthop is in a different vrf tell us\n"
+      "The nexthop-vrf Name\n")
+{
+	struct pbr_map_sequence *pbrms = VTY_GET_CONTEXT(pbr_map_sequence);
+	struct vrf *vrf;
+	struct nexthop nhop;
+	struct nexthop *nh;
+	struct pbr_event *pbre;
+
+	if (pbrms->nhgrp_name) {
+		vty_out(vty,
+			"Please unconfigure the nexthop group before adding an individual nexthop");
+		return CMD_WARNING;
+	}
+
+	if (name)
+		vrf = vrf_lookup_by_name(name);
+	else
+		vrf = vrf_lookup_by_id(VRF_DEFAULT);
+
+	if (!vrf) {
+		vty_out(vty, "Specified: %s is non-existent\n", name);
+		return CMD_WARNING;
+	}
+
+	memset(&nhop, 0, sizeof(nhop));
+	nhop.vrf_id = vrf->vrf_id;
+
+	if (addr->sa.sa_family == AF_INET) {
+		nhop.gate.ipv4.s_addr = addr->sin.sin_addr.s_addr;
+		if (intf) {
+			nhop.type = NEXTHOP_TYPE_IPV4_IFINDEX;
+			nhop.ifindex = ifname2ifindex(intf, vrf->vrf_id);
+			if (nhop.ifindex == IFINDEX_INTERNAL) {
+				vty_out(vty,
+					"Specified Intf %s does not exist in vrf: %s\n",
+					intf, vrf->name);
+				return CMD_WARNING;
+			}
+		} else
+			nhop.type = NEXTHOP_TYPE_IPV4;
+	} else {
+		memcpy(&nhop.gate.ipv6, &addr->sin6.sin6_addr, 16);
+		if (intf) {
+			nhop.type = NEXTHOP_TYPE_IPV6_IFINDEX;
+			nhop.ifindex = ifname2ifindex(intf, vrf->vrf_id);
+			if (nhop.ifindex == IFINDEX_INTERNAL) {
+				vty_out(vty,
+					"Specified Intf %s does not exist in vrf: %s\n",
+					intf, vrf->name);
+				return CMD_WARNING;
+			}
+		} else
+			nhop.type = NEXTHOP_TYPE_IPV6;
+	}
+
+	if (pbrms->nhg)
+		nh = nexthop_exists(pbrms->nhg, &nhop);
+	else {
+		char buf[100];
+
+		if (no) {
+			vty_out(vty, "No nexthops to delete");
+			return CMD_WARNING;
+		}
+
+		pbrms->nhg = nexthop_group_new();
+		pbrms->internal_nhg_name =
+			XSTRDUP(MTYPE_TMP,
+				pbr_nht_nexthop_make_name(pbrms->parent->name,
+							  pbrms->seqno,
+							  buf));
+		nh = NULL;
+	}
+
+	if (no) {
+		if (nh) {
+			// nexthop_del(pbrms->nhg, nh);
+			// nexthop_free(nh);
+			pbre = pbr_event_new(PBR_MAP_NEXTHOP_DELETE,
+					     pbrms->parent->name);
+			pbre->seqno = pbrms->seqno;
+			pbr_event_enqueue(pbre);
+		}
+	} else if (!nh) {
+
+		if (pbrms->nhg->nexthop) {
+			vty_out(vty,
+				"If you would like more than one nexthop please use nexthop-groups");
+			return CMD_WARNING;
+		}
+
+		/* must be adding new nexthop since !no and !nexthop_exists */
+		nh = nexthop_new();
+
+		memcpy(nh, &nhop, sizeof(nhop));
+		nexthop_add(&pbrms->nhg->nexthop, nh);
+
+		pbre = pbr_event_new(PBR_MAP_NEXTHOP_ADD, pbrms->parent->name);
+		pbre->seqno = pbrms->seqno;
+		pbr_event_enqueue(pbre);
+	}
+
+	return CMD_SUCCESS;
+}
+
+DEFPY (pbr_table_range,
+       pbr_table_range_cmd,
+       "[no]$no pbr table range (10000-65535)$start (11000-65535)$end",
+       NO_STR
+       "Policy based routing\n"
+       "Policy based routing table\n"
+       "Table range\n"
+       "Initial value of range\n"
+       "Final value of range\n")
+{
+	if (no)
```

Comment:
what is your strategy regarding integration of table manager range from zebra ? ( https://github.com/FRRouting/frr/pull/1824 ).
I have nothing against that choice for now ( the table range is very wide).
But I think we should specify the next step:
what about : pbr table range fromzebra [SIZEREQUESTED]

Reply:
I will update this in the future.  Not going to happen right now.

Reply:
so you mean the show running will change ?
or will you keep the default behaviour ( that is to say using hard set values for table manager ?)